src/ExpressionLanguage/ExpressionFunction/GraphQL/Mutation.php

```diff
@@ -8,11 +8,32 @@
 
 final class Mutation extends ExpressionFunction
 {
-    public function __construct($name = 'mutation')
+    public const NAME = 'mutation';
+
+    public function __construct($name = self::NAME)
     {
         parent::__construct(
             $name,
-            fn ($alias, $args = '[]') => "$this->globalVars->get('mutationResolver')->resolve([$alias, $args])"
+            function (string $alias, ...$args) {
+                $count = count($args);
+
+                // TODO: remove the following if-else-block in 1.0
+                if (1 === $count && '$' !== $args[0][0]) {
```

Comment:
`$args[0][0]` can be unset, so should we not test it to avoid notice ?

Reply:
`$args[0][0]` can never be unset. The `1 === $count` makes sure that `$args[0]` is set. Furthermore all items in the `$args` array are always non-empty strings, which means `$args[0][0]` points to the first letter of the string and is never unset.

Now a little explanation, why I did this check. We are changing the signature of the `mutation` function.

old: `mutation(string $alias, array $args = [])`
new: `mutation(string $alias, ...$args)`

The` if-block` tries to guess what signature user currently tries to use. If he calls the function with only 1 argument, which is an array (`'[' === $args[0][0]`), then it considers it old signature call, otherwise it uses the new signature.

The only drawback here is the case, where user tries to use the new signature and sends 1 array-argument:
`mutation("my_resolver", ['name', 'lastname'])`
So the bundle will think it's and old signature and will try to send `name` and `lastname` as 2 separate arguments to the resolver.

Reply:
yes I agree, I don't see any easy way to make the diffence between old and new signature in this case without using reflection.

Reply:
@mcg-web so what do you think I should do? Should I leave it as is?